sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -311,6 +311,67 @@ def predict(self, X):
         return X[:, 0]
 
 
+class OneClassSampleErrorClassifier(BaseBadClassifier):
+    def fit(self, X, y, sample_weight=None):
+        # Convert data
+        X, y = check_X_y(
+            X, y, accept_sparse=("csr", "csc"), multi_output=True, y_numeric=True
+        )
+
+        cls, y = np.unique(y, return_inverse=True)
+        nb_cls = cls.shape[0]
+        if nb_cls < 2:
+            raise ValueError("normal class error")
+
+        # find the number of class after trimming
+        if sample_weight is not None:
+            nb_cls = cls.shape[0]
+            if isinstance(sample_weight, np.ndarray) and len(sample_weight) > 0:
+                nb_cls = np.count_nonzero(np.bincount(y, sample_weight))
+            if nb_cls < 2:
+                raise ValueError("Nonsensical Error")
```

Comment:
From what I can tell, this it the difference between `OneClassSampleErrorClassifier` and `OneClassSampleErrorClassifierPredict`. Can we merge `OneClassSampleErrorClassifierPredict` into `OneClassSampleErrorClassifier` and have an `__init__` flag that triggers this error:

```python
class OneClassSampleErrorClassifier(...):
    def __init__(self, raise_when_single_class=False):
        self.raise_when_single_class_left = raise_when_single_class

    def fit(self, X, y):
        ...
            if nb_cls < 2:
                if self.raise_when_single_class:
                    raise ValueError(...)
                self.has_single_class_ = True
```